networkx/algorithms/bipartite/tests/test_generators.py

```diff
@@ -398,3 +432,31 @@ def test_gnmk_random_graph_complete(self):
         assert set(range(n)) == X
         assert set(range(n, n + m)) == Y
         assert edges == len(list(G.edges()))
+
+    def test_gnmk_random_graph_directed(self):
+        n = 10
+        m = 20
+        edges = 100
+        # set seed because sometimes it is not connected
+        # which raises an error in bipartite.sets(G) below.
+        G = gnmk_random_graph(n, m, edges, seed=1234, directed=True)
+        assert len(G) == n + m
+        assert nx.is_bipartite(G)
+        assert nx.is_directed(G)
+        X, Y = nx.algorithms.bipartite.sets(G)
+        # print(X)
+        assert set(range(n)) == X
+        assert set(range(n, n + m)) == Y
+        assert edges == len(list(G.edges()))
+
+    def test_gnmk_random_graph_star(self):
+        n = 1
+        m = 20
+        edges = 15
+        # set seed because sometimes it is not connected
+        # which raises an error in bipartite.sets(G) below.
+        # Remove set size test, as resulting graph should be disconnected for these values
+        G = gnmk_random_graph(n, m, edges, seed=1234)
+        assert len(G) == n + m
+        assert nx.is_bipartite(G)
+        assert edges == len(list(G.edges()))
```

Comment:
I'd vote to remove the comments here as I don't think they're necessary. Also I'd vote to rename the `edges` variable to something else (maybe `num_edges` or `k`)... the `assert edges ==` check is awkward with the current variable name (also there's no need to create a list: `len(list(G.edges()))` can be simplified to `len(G.edges)`)